### PR TITLE
[CI] Allow to choose nuget publishing feeds based in the repo id.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

**Review by chamons — APPROVED** [review]

File: tools/devops/automation/templates/release/publish-nugets.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    set -x
+    set -e
+    make
+    var=$(make -C $(Build.SourcesDirectory)/xamarin-macios/tools/devops print-variable VARIABLE=$VAR_NAME)
+    var=${var#*=}
+    echo "##vso[task.setvariable variable=Credential]$var"
+  displayName: 'Retrieve nuget feed'
+  name: ConfigureNuget
+  env:
+    ${{ if not(contains (variables['Build.Repository.ID'], 'private')) }}:
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/release/publish-nugets.yml:16]:
```suggestion
    ${{ if not(contains (variables['Build.DefinitionName'], 'private')) }}:
```

File: tools/devops/automation/templates/release/publish-nugets.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      VAR_NAME=NUGET_PUBLIC_FEED
+    ${{ if contains (variables['Build.Repository.ID'], 'private') }}:
```
Comment (mandel-macaque) [inline tools/devops/automation/templates/release/publish-nugets.yml:18]:
```suggestion
    ${{ if contains (variables['Build.DefinitionName'], 'private') }}:
```

File: tools/devops/automation/templates/release/publish-nugets.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    ${{ if not(contains (variables['Build.DefinitionName'], 'private')) }}:
+      VAR_NAME=NUGET_PUBLIC_FEED
+    ${{ if contains (variables['Build.DefinitionName'], 'private') }}:
+      VAR_NAME=NUGET_PRIVATE_FEED
+
+- task: NuGetAuthenticate@0
+  displayName: 'Authenticate dotnet-tool feed'
+  inputs:
+    nuGetServiceConnections: $(ConfigureNuget.Credential)
+  condition: contains (variables['Build.Repository.ID'], 'private')
```
Comment (dalexsoto) [inline tools/devops/automation/templates/release/publish-nugets.yml:25]: @mandel-macaque should we switch this to use `Build.DefinitionName`?

```suggestion
  condition: contains (variables['Build.DefinitionName'], 'private')
```

Comment (mandel-macaque) [reply tools/devops/automation/templates/release/publish-nugets.yml:25]: yes, I missed this one. nice catch

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**